capa/helpers.py

```diff
@@ -76,13 +86,46 @@ def load_json_from_path(json_path: Path):
         try:
             report_json = compressed_report.read()
         except gzip.BadGzipFile:
-            report = json.load(json_path.open(encoding="utf-8"))
+            report = msgspec.json.decode(json_path.read_text(encoding="utf-8"))
         else:
-            report = json.loads(report_json)
+            report = msgspec.json.decode(report_json)
     return report
 
 
+def load_jsonl_from_path(jsonl_path: Path) -> Iterator[Dict]:
+    def decode_json_lines(fd: Union[BinaryIO, gzip.GzipFile]):
+        for line in fd:
+            try:
+                line_s = line.strip().decode()
+                obj = msgspec.json.decode(line_s)
+                yield obj
+            except ValueError:
+                # ignore erroneous lines
+                continue
+
+    try:
+        with gzip.open(jsonl_path, "rb") as fg:
+            yield from decode_json_lines(fg)
+    except gzip.BadGzipFile:
+        with jsonl_path.open(mode="rb") as f:
+            yield from decode_json_lines(f)
+
+
+def load_one_jsonl_from_path(jsonl_path: Path):
+    # this loads one json line to avoid the overhead of loading the entire file
+    with jsonl_path.open(mode="rb") as f:
+        line = next(iter(f))
+        line = msgspec.json.decode(line.decode(errors="ignore"))
+    return line
+
+
 def get_format_from_report(sample: Path) -> str:
+    if sample.name.endswith((".log", "log.gz")):
+        line = load_one_jsonl_from_path(sample)
```

Comment:
does this work for raw and compressed files?

Reply:
No, I have added this now however.